### PR TITLE
Adjust diesel fuel consumption

### PR DESCRIPTION
**Pull request by peternewell**
Adjust fuel consumption so that it will operate whilever the diesel engine is running.

https://bugs.launchpad.net/or/+bug/1953251

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerSupplies/DieselEngine.cs [hunk 1]
```diff
@@ -1085,7 +1085,8 @@ public void Update(float elapsedClockSeconds)
             if ((State != DieselEngineState.Starting) && (RealRPM == 0f))
                 State = DieselEngineState.Stopped;
 
-            if ((State == DieselEngineState.Stopped) || (State == DieselEngineState.Stopping) || ((State == DieselEngineState.Starting) && (RealRPM < StartingRPM)))
+            // fuel consumption will occur when engine is running above the starting rpm
+            if ((State == DieselEngineState.Stopped) || ((State == DieselEngineState.Stopping) && (RealRPM < StartingRPM)) || ((State == DieselEngineState.Starting) && (RealRPM < StartingRPM)))
```
Comment (pzgulyas) [inline Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerSupplies/DieselEngine.cs:1089]: && takes precedence over ||, and == takes precedence over both of these, so brackets are totally unnecessary in this line except the if()'s brackets. See: [Operator precedence](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/#operator-precedence)

**Review by pzgulyas — APPROVED** [review]

File: Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerSupplies/DieselEngine.cs (hunk 1 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerSupplies/DieselEngine.cs]: Whilst I agree that it is possible to remove the brackets, sometimes I prefer a long hand approach as I believe that it makes the code more readable for all.

So I have removed some of the brackets and left a couple on this basis.

Thanks for highlighting this.

Could you please approve this PR now.

**Review by pzgulyas — APPROVED** [review]
No question I give my thubms-up to you!

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs [hunk 2]
```diff
@@ -662,6 +650,24 @@ protected override void UpdateTractiveForce(float elapsedClockSeconds, float t,
                     w = 0;
                 AverageForceN = w * AverageForceN + (1 - w) * TractiveForceN;
             }
+
+            // Calculate fuel consumption will occur unless diesel engine is stopped
+            if (LocomotivePowerSupply.MainPowerSupplyOn)
```
Comment (Sharpe49) [inline Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs:655]: Main power supply on means that power is available for traction, not that the engine is running.
You should use diesel engine state.

**Review by Sharpe49 — CHANGES_REQUESTED** [review]

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs (hunk 2 above)
Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs]: Ok, I have removed this condition as the diesel engine state should manage diesel fuel flow.

Comment (Sharpe49) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs]: The functionnality is correct, but the indentation needs to be fixed.

Comment (peternewell) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs]: Indentation fixed

**Review by Sharpe49 — APPROVED** [review]

File: Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs (hunk 2 above)
Comment (Sharpe49) [reply Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs]: Thank you. It's approved.

**Comment by peternewell** [comment]
Thanks for that.

**Pull request merged**